src/components/forms/ItemQuantityForm.tsx

```diff
@@ -0,0 +1,60 @@
+import { FormEvent, useState } from "react";
+import { ListItem } from "../../api";
+
+interface ItemQuantityFormProps {
+	saveItemQuantity: (quantity: number) => void;
+	item: ListItem;
+}
+
+const ItemQuantityForm: React.FC<ItemQuantityFormProps> = ({
+	saveItemQuantity,
+	item,
+}) => {
+	// A state variable to store the item quantity.
+	const [itemQuantity, setItemQuantity] = useState<number>(item?.itemQuantity);
```

Comment:
suggestion: since the default quantity is 1 for all items it may be nice to have this value be set to that so on the adding item form it shows what it will be for the user.

```suggestion
const [itemQuantity, setItemQuantity] = useState<number>(item ? item.itemQuantity : 1);
```

Reply:
Great suggestion! I'm a little curious why item could be undefined in the first place 👀  do we need the question mark at all?

Reply:
+1

Reply:
I needed the "?" to make it a nullable type because if there was no items on the list, an error would be thrown in the browser about "item" being null or undefined.

.................Because I was overthinking a logical operator for useState<number>(item.itemQuantity || 1) . 🤣 🤣 🤣 🤣 🤣

Reply:
> suggestion: since the default quantity is 1 for all items it may be nice to have this value be set to that so on the adding item form it shows what it will be for the user.

Yes please! Thank you for noticing! I was trying to do that with some logical operators. I should have just tried out ternary operators!

Reply:
> +1

💯